File: Garage_Rental.Infra/Service/RoleService.cs
using Garage_Rental.Core.Data;
using Garage_Rental.Core.Repository;
using Garage_Rental.Core.Service;
using System;
using System.Collections.Generic;
using System.Text;


namespace Garage_Rental.Infra.Service
{
    public class RoleService : IGenericService<Role>
    {
        private readonly IGenericRepository<Role> _roleRepository;

        public RoleService(IGenericRepository<Role> RoleRepository)
        {
            _roleRepository = RoleRepository;
        }

        public bool Create(Role t)
        {
            return _roleRepository.Create(t);
        }

        public void Delete(int id)
        {
            _roleRepository.Delete(id);
        }

        public List<Role> GetAll()
        {
            return _roleRepository.GetAll();
        }

        public Role GetById(int id)
        {
            return _roleRepository.GetById(id);
        }

        public void Update(Role t)
        {
            _roleRepository.Update(t);
        }
    }
}

File: FrontEnd/GarageRentalFrontEnd/src/app/Services/role.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { environment } from 'src/environments/environment';
import { Role } from '../Models/role.model';

@Injectable({
  providedIn: 'root'
})
export class RoleService {


 baseApiUrl:string = environment.baseApiUrl;

  constructor(private http:HttpClient) { }


  getRoleList(): Observable<Role[]>{
    return this.http.get<Role[]>(this.baseApiUrl +'Role/GetAll');
  }

  getRole(id:string):Observable<Role>{
    return this.http.get<Role>(this.baseApiUrl +'Role/GetById?id=' + id);
  }

  addRole(addRoleReq:Role):Observable<Role>{
    return this.http.post<Role>(this.baseApiUrl +'Role/Create',addRoleReq);
  }

  updateRole(id:number,updateRolereq:Role):Observable<Role>{
    return this.http.post<Role>(this.baseApiUrl +'Role/Update?id='+ id ,updateRolereq)
  }
  deleteRole(id:number):Observable<Role>{
    return this.http.delete<Role>(this.baseApiUrl +'Role/Delete?id=' + id);
  }
}
